compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -503,15 +503,16 @@ class Typer extends Namer
       // Convert a reference `f` to an extension method select `p.f`, where
       // `p` is the closest enclosing extension parameter, or else convert to `this.f`.
       val xmethod = ctx.owner.enclosingExtensionMethod
-      val qualifier =
-        if xmethod.exists then untpd.ref(xmethod.extensionParam.termRef)
-        else untpd.This(untpd.EmptyTypeIdent)
-      val selection = untpd.cpy.Select(tree)(qualifier, name)
-      val result = tryEither(typed(selection, pt))((_, _) => fail)
-      def canAccessUnqualified(sym: Symbol) =
-        sym.is(ExtensionMethod) && (sym.extensionParam.span == xmethod.extensionParam.span)
-      if !xmethod.exists || result.tpe.isError || canAccessUnqualified(result.symbol) then
-        result
+      if xmethod.exists then
```

Comment:
I would have thought that the code here would be executed only rarely.

Reply:
For a simple program (see below), the speedup is about 24% (34ms/op VS 45ms/op):

```Scala
class MyInt(val x: Int) {
  def eq(that: MyInt): Boolean = this.x == that.x
}

class Test {
  def foo(x: MyInt, y: MyInt): Boolean = x.eq(y)

  val a = MyInt(2)
  val b = MyInt(3)
  foo(a, b)
}
```

Reply:
Interesting. I just found out why: The code in question (including the expensive tryEither) is executed twice, once for each MyInt(...). I had not thought of that. So, yes, because of constructor syntax, failures in findRef are now not that uncommon. So this is a definite improvement. Unfortunately, the benchmarks are too noisy to see this immediately.